### DISCUSSION TITLE
Intel SYCL standard runtime support with oneAPI

### DISCUSSION THREAD

**Discussion by abhilash1910** [karpathy/llm.c#572] [Ideas]
Firstly thanks @karpathy  for creating the repository.
This is a discussion thread regarding a support /collaboration with standard SYCL runtime from Intel .
[SYCL](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html) being a hardware agnostic language, can be useful for porting and running different kernels across different runtimes . Since there was a mention by @karpathy  that this repository should be only Cuda specific, there is a proposal to have an official SYCL implementation from Intel (oneapi) .
Currently I will be heading the development of the SYCL /oneapi compiler backend from Intel for this project and wanted to know :

- Is it possible to have a SYCL branch from this repository which would be contributed and maintained by myself and Intel colleagues?
- If not, can the new repository for sycl oneapi support be linked in the readme of this repository ? 
- How to best collaborate regarding unified SYCL runtime across different vendors for the kernels ? (Through devices from Intel etc)
 - Best way to benchmark accelerator interfaces such as Nvidia cuda/cudnn vs sycl/onednn  and blas for performance ?
 
 I am open to suggestions and greatly appreciate the effort on this repository.

**Comment by abhilash1910** [comment]
Request : https://github.com/karpathy/llm.c/pull/579

**Comment by karpathy** [comment]
Hi @abhilash1910 . Great, excited to hear! I won't accept PRs into this repo but I'm very happy to link to yours in the "notable forks" section. Or answer any questions you might have or etc of course. Cheers, -A

**Reply by abhilash1910** [reply]
Thanks @karpathy , yes it would be best if you could link the repository : https://github.com/abhilash1910/llm.sycl/ 
Should I create a PR to add the repo link ?
We plan to have the same codebase inside intel/ as well and would ping/PR when that is done from our side.
Thanks again for the quick response, appreciated 👍🏻

**Reply by abhilash1910** [reply]
PR for readme update: https://github.com/karpathy/llm.c/pull/581

**Reply by abhilash1910** [reply]
Hi @karpathy , I had a doubt, will this repo only support gpt2 , or is there a plan to add on llama and other models as well ?
Also could you please help with #581 . Many thanks. 
PS: Also an inprogress development work is ongoing for habana accelerators which I will link and let you know.

**Reply by abhilash1910** [reply]
Hi @karpathy  could you help review #581  and #668  as both accelerator codes are in ready state for single card runs . More details will be followed on the separate forks.  Includes Intel SYCL and Habana Gaudi accelerators.
